datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -411,11 +411,11 @@ pub async fn from_substrait_rel(
                         from_substrait_rex(ctx, e, input.clone().schema(), extensions)
                             .await?;
                     // if the expression is WindowFunction, wrap in a Window relation
-                    if let Expr::WindowFunction(_) = x.as_ref() {
+                    if let Expr::WindowFunction(_) = &x {
```

Comment:
I hoped clippy can find things like this 🤔

Reply:
For some reason, to convert an `Arc<T>` to ` &T` you can't just do 

```rust
let my_ref = &my_arc;
```

Rust seems to require you to do 

```rust
let my_ref = my_arc.as_ref();
```

I don't really know the nuances as to why.

After this change, `x` is no longer an `Arc<Expr>` it is only an `Expr` which Rust is happier to auto-reference